app/components/add-cron-job.js

```diff
@@ -34,19 +34,9 @@ export default Component.extend({
     });
   },
 
-  search: task(function* (query) {
-    yield timeout(config.intervals.searchDebounceRate);
-    let branchNames = this.get('currentCronJobsBranches');
-    let branches = yield this.store.query('branch', {
-      repository_id: this.repository.id,
-      data: {
-        name: query,
-        sort_by: 'name',
-        limit: 10,
-        exists_on_github: true
-      }
-    });
-    return branches.reject(branch => (branchNames.includes(branch.name)));
+  search: task(function (query) {
+    const branchNames = this.get('currentCronJobsBranches');
+    return this.searchBranch(this.repository.id, query, branchNames);
```

Comment:
`task` is expected to be a generator function. It doesn't make any sense otherwise, because regular function can not be concurrent and can not be interrupted. Although, tasks can run other tasks. So if you'd convert `searchBranch` to be a `task`, you'd be able to call it here:
```js
search: task(function* (query) {
  const branchNames = this.currentCronJobsBranches;
  const searchResults = yield this.searchBranch.perform(this.repository.id, query, branchNames);
  return searchResults;
})
```